### PR TITLE
Update controller-runtime, controller-tools, k8s-client and kustomize

### LINKED ISSUE #511: Upgrade the versions of tools in CAPG
/kind feature

**Describe the solution you'd like**

CAPG still uses older versions of controller-runtime, K8s-client, controller-tools and Kustomize. It will be better if we can upgrade it to the latest version, to avoid any future inconvenience, and to be in parity with CAPI & other CAP providers.
- upgrade controller-runtime to 0.11.0
- upgrade K8s-client to 1.23
- upgrade controller-tools 0.8.0
- upgrade Kustomize to v4.4.0

### PR DESCRIPTION
**Pull request by jayesh-srivastava**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Bumps controller-runtime to 0.11.0 
Bumps controller-tools to 0.8.0
Bumps k8s-client to 1.23.1
Bumps customize to 4.4.0

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #511 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:
```
Bumps controller-runtime to 0.11.0 
Bumps controller-tools to 0.8.0
Bumps k8s-client to 1.23.1
Bumps customize to 4.4.0
```

### CODE REVIEW

File: scripts/ci-conformance.sh [hunk 1]
```diff
@@ -47,8 +47,8 @@ export GCP_NETWORK_NAME=${GCP_NETWORK_NAME:-"${TEST_NAME}-mynetwork"}
 GCP_B64ENCODED_CREDENTIALS=$(base64 -w0 "$GOOGLE_APPLICATION_CREDENTIALS")
 export GCP_B64ENCODED_CREDENTIALS
 export KUBERNETES_MAJOR_VERSION="1"
-export KUBERNETES_MINOR_VERSION="22"
-export KUBERNETES_PATCH_VERSION="6"
+export KUBERNETES_MINOR_VERSION="23"
+export KUBERNETES_PATCH_VERSION="1"
```
Comment (cpanato) [inline scripts/ci-conformance.sh:51]: you can bump to 1.23.3, but I would keep that in 1.22.6 for now

File: scripts/ci-e2e.sh [hunk 2]
```diff
@@ -47,8 +47,8 @@ export GCP_NETWORK_NAME=${GCP_NETWORK_NAME:-"${TEST_NAME}-mynetwork"}
 GCP_B64ENCODED_CREDENTIALS=$(base64 "$GOOGLE_APPLICATION_CREDENTIALS" | tr -d '\n')
 export GCP_B64ENCODED_CREDENTIALS
 export KUBERNETES_MAJOR_VERSION="1"
-export KUBERNETES_MINOR_VERSION="22"
-export KUBERNETES_PATCH_VERSION="6"
+export KUBERNETES_MINOR_VERSION="23"
+export KUBERNETES_PATCH_VERSION="1"
```
Comment (cpanato) [inline scripts/ci-e2e.sh:51]: same: you can bump to 1.23.3, but I would keep that in 1.22.6 for now

File: test/e2e/config/gcp-ci.yaml [hunk 3]
```diff
@@ -65,13 +65,13 @@ providers:
       targetName: "cluster-template-conformance-ci-artifacts.yaml"
 
 variables:
-  KUBERNETES_VERSION: "${KUBERNETES_VERSION:-v1.22.6}"
+  KUBERNETES_VERSION: "${KUBERNETES_VERSION:-v1.23.1}"
```
Comment (cpanato) [inline test/e2e/config/gcp-ci.yaml:68]: same: you can bump to 1.23.3, but I would keep that in 1.22.6 for now

File: test/e2e/config/gcp-ci.yaml [hunk 4]
```diff
@@ -65,13 +65,13 @@ providers:
...
   EXP_CLUSTER_RESOURCE_SET: "true"
   # Cluster Addons
   CNI: "${PWD}/test/e2e/data/cni/calico/calico.yaml"
   GCP_CONTROL_PLANE_MACHINE_TYPE: n1-standard-2
   GCP_NODE_MACHINE_TYPE: n1-standard-2
-  CONFORMANCE_CI_ARTIFACTS_KUBERNETES_VERSION: "v1.22.6"
+  CONFORMANCE_CI_ARTIFACTS_KUBERNETES_VERSION: "v1.23.1"
```
Comment (cpanato) [inline test/e2e/config/gcp-ci.yaml:74]: same: you can bump to 1.23.3, but I would keep that in 1.22.6 for now

**Review by cpanato — COMMENTED** [review]
some few comments

**Comment by cpanato** [comment]
did not see this `Bumps controller-runtime to 0.11.0`

**Comment by jayesh-srivastava** [comment]
> did not see this `Bumps controller-runtime to 0.11.0`

@cpanato I saw in the codebase that this was already updated. Should I remove this from the release notes? It was mentioned in the issue so I mentioned it here too.

File: scripts/ci-conformance.sh (hunk 1 above)
Comment (jayesh-srivastava) [reply scripts/ci-conformance.sh:51]: I saw CAPZ updated the k8s version to 1.23. So should I update in CAPG or leave it to 1.22.6

File: scripts/ci-e2e.sh (hunk 2 above)
Comment (cpanato) [reply scripts/ci-e2e.sh:51]: we dont have prebuit images for 1.23.1 only for 1.23.3, can you please update for that?

**Comment by jayesh-srivastava** [comment]
@cpanato Can you please help me with the failing test? Can't seem to understand it?

File: scripts/ci-e2e.sh (hunk 2 above)
Comment (jayesh-srivastava) [reply scripts/ci-e2e.sh:51]: Sure!

**Comment by jayesh-srivastava** [comment]
cc @cpanato

**Review by cpanato — APPROVED** [review]
great job! thanks @jayesh-srivastava 

/approve
/lgtm

**Pull request merged**